src/main/java/step3/Car.java

```diff
@@ -1,13 +1,39 @@
 package step3;
 
-public class Car {
+public class Car implements Comparable<Car> {
+  private final String name;
   private int distance;
+  private final CarMovingStartegy carMovingStartegy;
+
+  public Car(CarMovingStartegy carMovingStartegy) {
+    this.name = "";
+    this.carMovingStartegy = carMovingStartegy;
```

Comment:
요구사항이 변경되었는데, 이 생성자 필요할까요? 
만약 기존 서비스 유지를 위해 필요하다면 `this` 키워드를 사용하여 생성자를 호출하면 어떨까요?

```suggestion
    this("", carMovingStartegy);
```

이후에 생성자의 개수가 많아졌을 땐 빌더 패턴을 사용할 수도 있어요. 어떤 패턴인지 가볍게 공부해보셔도 좋을 것 같아요.